batch.py

```diff
@@ -776,8 +777,9 @@ def manage_concurrency_for_job(job: hb.job.Job):
             prepare_input_job = batch.new_python_job(f'Prepare inputs for: {gene}')
             manage_concurrency_for_job(prepare_input_job)
             copy_common_env(prepare_input_job)
-            if gene_dict[gene]['plink_job'] is not None:
-                prepare_input_job.depends_on(gene_dict[gene]['plink_job'])
+            dependency = dependencies_dict.get(gene)
+            if dependency:
+                prepare_input_job.depends_on(dependency)
```

Comment:
You could further condense this (but some people hate the syntax)

```suggestion
            if dependency := dependencies_dict.get(gene):
                prepare_input_job.depends_on(dependency)
```

Reply:
Perfect! I always forget about the walrus ;)